hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/EnhancementAsProxyLazinessInterceptor.java

```diff
@@ -215,23 +216,7 @@ public Object forceInitialize(Object target, String attributeName, SharedSession
 
 		if ( isTemporarySession ) {
 			// Add an entry for this entity in the PC of the temp Session
-			session.getPersistenceContextInternal().addEntity(
-					target,
-					org.hibernate.engine.spi.Status.READ_ONLY,
-					// loaded state
-					ArrayHelper.filledArray(
-							LazyPropertyInitializer.UNFETCHED_PROPERTY,
-							Object.class,
-							persister.getPropertyTypes().length
-					),
-					entityKey,
-					persister.getVersion( target ),
-					LockMode.NONE,
-					// we assume an entry exists in the db
-					true,
-					persister,
-					true
-			);
+			session.getPersistenceContext().addEnhancedProxy( entityKey, asPersistentAttributeInterceptable( target ) );
```

Comment:
Is this change actually tested? I did not see anything obvious in the tests that would be creatiung temporary sessions. Are the test entities enhanced or something?

Reply:
Though the PR doesn't involve bytecode enhancement we have tests that rely on this code. There was a failing test due to to the fact that we were adding the uninitialized bytecode proxy (along with an array of `UNFETCHED_PROPERTY`s as the state) as an actual entity here, marking the holder as `INITIALIZED`. To me this looked wrong, and `addEnhancedProxy()` seemed like the perfect method to use instead.

Reply:
You mean this test started failing after the other changes in the PR?  Just trying to understand :)

Reply:
Ok, that makes sense.